Home assignment -webTables

elmirakamil opened the pull request:
Please review my homework for web tables.

Thank you!

elmirakamil committed:
Home assignment -webTables

bondar-artem commented on tests/webTables.spec.ts:
```diff
@@ -0,0 +1,182 @@
+    //8. Add assertion that "surgery" was changed to "dermatology" in the list of specialties
+    const specialtyInput = page.locator('#name')
+    await expect(specialtyInput).toHaveValue('surgery')
+    await specialtyInput.clear()
```
`clear()` is not needed, method `fill()` will handle it

bondar-artem commented on tests/webTables.spec.ts:
```diff
@@ -0,0 +1,182 @@
+    }
+
+    await expect(specialtyInput).toHaveValue('dermatology')
+    await specialtyInput.clear()
```
`clear()` is not needed, method `fill()` will handle it

bondar-artem commented on tests/webTables.spec.ts:
```diff
@@ -0,0 +1,182 @@
+test('validate specialty lists', async ({ page }) => {
+    //1. Select the SPECIALTIES menu item in the navigation bar
+    await page.getByText('Specialties').click()
+    await page.waitForSelector('tbody')
```
this wait is not needed, because on the next step the `click()` method will wait for "Add" button automatically

bondar-artem commented on tests/webTables.spec.ts:
```diff
@@ -0,0 +1,182 @@
+
+    //4. Select the VETERINARIANS menu item in the navigation bar, then select "All"
+    await page.getByText('Veterinarians').click()
+    await page.locator(".dropdown-menu").getByText(" All").click()
```
Earlier in your code in line 122, you have used a simpler version of the locator than this version. Why?

bondar-artem commented on tests/webTables.spec.ts:
```diff
@@ -0,0 +1,182 @@
+    await page.locator('tbody tr').last().getByRole('button', { name: "Delete" }).click()
+    //11. Navigate to VETERINARIANS page. Add assertion that "Sharon Jenkins" has no specialty assigned
+    await page.getByText('Veterinarians').click()
+    await page.locator(".dropdown-menu").getByText(" All").click()
```
Earlier in your code in line 122, you have used a simpler version of the locator than this version. Why?

bondar-artem reviewed (commented):
Great job!
To be honest I am pleasantly surprised to see this progress. Web tables is one of the most difficult topics. Almost nothing to add from my side, just a couple of comments.

elmirakamil commented:
I gave more than 2 weeks to do it. Still getting some parts :) Thank you for your comments will work on them

elmirakamil committed:
Homework-webTables-version 2

elmirakamil commented:
I've addressed your comments, but I'm still struggling with staying focused on my tasks. I think the reason I made multiple changes to the locators is because I worked on them separately, which took more time. I will continue to improve my scripts. Thank you!

elmirakamil commented:
Appreciate it!
